doc/decisions/iterators.md

```diff
@@ -0,0 +1,52 @@
+# Iterators
+
+## Problem
+
+The internal iterator inside KeySets seems to cause more problems than it solves.
+
+## Constraints
+
+We could embed one such iterator into KeySets, but recommend that people use an 
+external instance (int) in new code.
+
+## Assumptions
+
+@raphi011 made benchmarks showing that external and internal iterators have
+about the same speed. So it should be a safe choice to not provide the internal
+iterators for 1.0 and only the external instead.
+
+## Considered Alternatives
+
+## Decision
+
+- remove all functions related to the internal iterator:
+  - ksRewind
```

Comment:
We should also add `keyRewindMeta` to the list of functions to remove

Reply:
what about `keyNextMeta()` / `keyCurrentMeta`?

Reply:
Yes, `keyRewindMeta()`, `keyNextMeta()` and `keyCurrentMeta()` should be removed. Both because of the iterator changes and because of the `keyMeta()` changes.